tests/backup/backup_test.go

```diff
@@ -2800,6 +2809,225 @@ var _ = Describe("{ShareBackupWithDifferentRoleUsers}", func() {
+		})
+
+		Step("Creating backup location and cloud setting", func() {
+			log.InfoD("Creating backup location and cloud setting")
+			for _, provider := range providers {
+				bucketName := fmt.Sprintf("%s-%s", provider, bucketNames[0])
+				cloudCredName = fmt.Sprintf("%s-%s", "cred", provider)
+				bkpLocationName = fmt.Sprintf("%s-%s-bl", provider, bucketNames[0])
+				cloudCredUID = uuid.New()
+				backupLocationUID = uuid.New()
+				backupLocationMap[backupLocationUID] = bkpLocationName
+				CreateCloudCredential(provider, cloudCredName, cloudCredUID, orgID)
+				// TODO remove time.sleep: PA-509
+				time.Sleep(time.Minute * 5)
+				CreateBackupLocation(provider, bkpLocationName, backupLocationUID, cloudCredName, cloudCredUID, bucketName, orgID, "")
+			}
+		})
+
+		Step("Register cluster for backup", func() {
+			CreateSourceAndDestClusters(orgID, "", "", ctx)
+			clusterStatus, clusterUid = Inst().Backup.RegisterBackupCluster(orgID, SourceClusterName, "")
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, "Verifying backup cluster")
+		})
+
+		Step("Taking backups of application for each user", func() {
+			log.InfoD("Taking backup of application for each user")
+			var sem = make(chan struct{}, 10)
+			var wg sync.WaitGroup
+			for i := 0; i < numberOfUsers; i++ {
+				sem <- struct{}{}
+				time.Sleep(3 * time.Second)
+				backupName := fmt.Sprintf("%s-%v", BackupNamePrefix, time.Now().Unix())
+				backupNames = append(backupNames, backupName)
+				wg.Add(1)
+				go func(backupName string) {
+					defer GinkgoRecover()
+					defer wg.Done()
+					defer func() { <-sem }()
+					err = CreateBackup(backupName, SourceClusterName, bkpLocationName, backupLocationUID, []string{bkpNamespaces[0]},
+						labelSelectors, orgID, clusterUid, "", "", "", "", ctx)
+					log.FailOnError(err, "Failed while trying to take backup of application- %s", bkpNamespaces[0])
+				}(backupName)
+			}
+			wg.Wait()
+			log.Infof("List of backups - %v", backupNames)
+		})
+
+		Step("Share backup with users with different access level", func() {
+			_, err = ShareBackupWithMultipleUsers(backupNames, users, ctx)
+			dash.VerifyFatal(err, nil, "Sharing backup with users")
+		})
+
+		Step("Removing backup location after sharing backup with all the users", func() {
+			err = DeleteBackupLocation(bkpLocationName, backupLocationUID, orgID)
+			dash.VerifySafely(err, nil, fmt.Sprintf("Deleting backup location %s", bkpLocationName))
+		})
+
+		Step("Adding new backup location to the cluster", func() {
+			for _, provider := range providers {
+				bucketName := fmt.Sprintf("%s-%s", provider, bucketNames[0])
+				newBkpLocationName = fmt.Sprintf("new-%s-%s-bl", provider, bucketNames[0])
+				newBackupLocationUID = uuid.New()
+				newBackupLocationMap[newBackupLocationUID] = newBkpLocationName
+				CreateBackupLocation(provider, newBkpLocationName, newBackupLocationUID, cloudCredName, cloudCredUID, bucketName, orgID, "")
```

Comment:
Can we add return to `CreateBackupLocation()`, let's be consistent across similar functions, I see that `DeleteBackupLocation()` returns error, I don't see a reason why `CreateBackupLocation()` should be written in same style

Reply:
done!!